### ISSUE TITLE
Create slack endpoint 

### ISSUE THREAD

**Issue by nathanielc** [influxdata/flux#1689]
Create a Slack endpoint as described here https://github.com/influxdata/flux/blob/master/docs/Outputs.md

Here are a list of configuration options for a slack endpoint:

* workspace - string - Name of the slack workspace to use if there are multiple. Defaults to empty string
* url - string - URL of the slack endpoint. No default value, this must be acquired as part of the slack API setup. This URL will be secret. Don't worry about secrets for the initial implementation.
* channel - string - Name of channel in which to post the message. No default
* username - string - Username posting the message
* message - string - The text to display 
* icon-emoji - string - Name of `:` enclose emoji to use as image of user when posting message.
* color - string - Color to give message: one of `good`, `warning`, and `danger`, or any hex rgb color value ex. `#439FE0`.

Of the above only `url` should be a parameter to the `endpoint` function, the rest of the options must be provided as the return of mapFn. This allows for all of those values to be dynamic based on the data, or static based on the needs of the user.

**Comment by nathanielc** [comment]
Kapacitor already has an isolated package for testing calls to this API endpoint here https://github.com/influxdata/kapacitor/tree/master/services/slack/slacktest

I recommend that code is copied into Flux for facilitating testing.

**Comment by docmerlin** [comment]
token needs to be another parameter?

**Comment by nathanielc** [comment]
@docmerlin No, turns out slack doesn't actually need a `token`. The auth token is part of the URL itself. SO the example in the Outputs.md doc is inaccurate in that regard, it was just demonstrating how such a parameter should be implemented.
